pkg/framework/proxies.go

```diff
@@ -0,0 +1,249 @@
+	return machineSet, nil
+}
+
+// DeployClusterProxy Deploys an HTTP proxy to the proxy node
+func DeployClusterProxy(c runtimeclient.Client) (*appsv1.Deployment, error) {
+	configMap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-conf",
+			Namespace: "default",
+		},
+		Data: map[string]string{
+			"squid.conf": squidConf,
+		},
+	}
+
+	c.Create(context.Background(), &configMap)
+
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-proxy",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app": "squid",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "squid",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "squid-proxy",
+					Namespace: "default",
+					Labels: map[string]string{
+						"app": "squid",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "conf",
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "squid-conf",
+									},
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "squid",
+							Image: "sameersbn/squid:3.5.27-2",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 80,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "conf",
+									ReadOnly:  false,
+									MountPath: "/etc/squid/squid.conf",
+									SubPath:   "squid.conf",
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{},
+						},
+					},
+					NodeSelector: map[string]string{
+						"proxy-host": "",
+					},
+					HostNetwork: true,
+				},
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: "Recreate",
+			},
+			Paused: false,
+		},
+	}
+	c.Create(context.Background(), deployment)
+	IsDeploymentAvailable(c, deployment.Name, "default")
+	return deployment, nil
+}
+
+// DestroyClusterProxy destroys the HTTP proxy and associated configmap
+func DestroyClusterProxy(c runtimeclient.Client) error {
+	configMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-conf",
+			Namespace: "default",
+		},
+	}
+	c.Delete(context.Background(), configMap)
+
+	return DeleteDeployment(c, &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "squid-proxy",
+			Namespace: "default",
+		},
+	})
+}
+
+// WaitForProxyInjectionSync waits for the deployment to sync with the state of the cluster-proxy
+func WaitForProxyInjectionSync(c runtimeclient.Client, name, namespace string, shouldBePresent bool) bool {
+	if err := wait.PollImmediate(RetryShort, WaitLong, func() (bool, error) {
```

Comment:
On a `WaitLong`, might want to reduce the retry, perhaps `RetryMedium` would be more appropriate here?